Mathlib/Analysis/NormedSpace/HahnBanach/Separation.lean

```diff
@@ -200,3 +202,103 @@ theorem iInter_halfspaces_eq (hs₁ : Convex ℝ s) (hs₂ : IsClosed s) :
   obtain ⟨l, s, hlA, hl⟩ := geometric_hahn_banach_closed_point hs₁ hs₂ h
   obtain ⟨y, hy, hxy⟩ := hx l
   exact ((hxy.trans_lt (hlA y hy)).trans hl).not_le le_rfl
+
+section RCLike
+
+open RCLike
+
+variable [RCLike 𝕜] [TopologicalSpace E] [AddCommGroup E] [TopologicalAddGroup E]
+  [Module 𝕜 E] [Module ℝ E] [ContinuousSMul 𝕜 E] [IsScalarTower ℝ 𝕜 E]
+
+/--Real linear extension of continuous extension of `LinearMap.extendTo𝕜'` -/
+noncomputable def LinTo𝕜' : (E →L[ℝ] ℝ) →ₗ[ℝ] (E →L[𝕜] 𝕜) :=
+  letI to𝕜 (fr : (E →L[ℝ] ℝ)) : (E →L[𝕜] 𝕜) :=
+    { toLinearMap := LinearMap.extendTo𝕜' fr
+      cont := show Continuous fun x ↦ (fr x : 𝕜) - (I : 𝕜) * (fr ((I : 𝕜) • x) : 𝕜) by fun_prop }
+  have h fr x : to𝕜 fr x = ((fr x : 𝕜) - (I : 𝕜) * (fr ((I : 𝕜) • x) : 𝕜)) := rfl
+  { toFun := to𝕜
+    map_add' := by intros; ext; simp [h]; ring
+    map_smul' := by intros; ext; simp [h, real_smul_eq_coe_mul]; ring }
+
+@[simp]
+lemma real_of_LinTo𝕜' (g : E →L[ℝ] ℝ) (x : E) :  re ((LinTo𝕜' g) x : 𝕜) = g x := by
+  have h g (x : E) : LinTo𝕜' g x = ((g x : 𝕜) - (I : 𝕜) * (g ((I : 𝕜) • x) : 𝕜)) := rfl
```

Comment:
This should be a lemma, possibly generated by `@[simps]` on the definition

Reply:
Tried @[simps] on the definition, and it fails, suggesting @[simps!]. When I tried that, it broke the definition and several things below.

Reply:
By the way, though, if I give the definition `@[simp]` this works. Is that an ok change?

Reply:
Actually, I just tried @[simps!] again, and it didn't break this time...and I have no idea why. I noticed that this generates a longer list of lemmas to apply in the call than if I stick with the single lemma I introduced. Is it better to use the auto generated lemmas if it leads to using more rewrites or is it better to introduce the new lemma. Thanks again!

Reply:
You can use `@[simps!?]` or `@[simps?]` to make it tell you what lemmas it generated. In this case, I think the one you wrote by hand is better.

Reply:
Cool. I rolled those changes back.